src/goto-instrument/contracts/inlining_decorator.cpp

```diff
@@ -0,0 +1,142 @@
+/*******************************************************************\
+
+Module: Utility functions for code contracts.
+
+Author: Remi Delmas, delmasrd@amazon.com
+
+Date: August 2022
+
+\*******************************************************************/
+
+#include "inlining_decorator.h"
+
+inlining_decoratort::inlining_decoratort(message_handlert &_wrapped)
+  : wrapped(_wrapped),
+    no_body_regex(std::regex(".*no body for function '(.*)'.*")),
+    missing_function_regex(
+      std::regex(".*missing function '(.*)' is ignored.*")),
+    recursive_call_regex(
+      std::regex(".*recursion is ignored on call to '(.*)'.*")),
+    not_enough_arguments_regex(
+      std::regex(".*call to '(.*)': not enough arguments.*"))
+{
+}
+
+void inlining_decoratort::match_no_body_warning(const std::string &message)
+{
+  std::smatch sm;
+  std::regex_match(message, sm, no_body_regex);
+  if(sm.size() == 2)
+  {
+    std::stringstream ss;
+    ss << sm[1];
+    no_body_set.insert(ss.str());
+  }
+}
+
+void inlining_decoratort::match_missing_function_warning(
+  const std::string &message)
+{
+  std::smatch sm;
+  std::regex_match(message, sm, missing_function_regex);
+  if(sm.size() == 2)
```

Comment:
Why 2? The size will return "number of capturing groups plus one", so I guess you need at least one(i.e., `sm.size() > 1`), right?

Reply:
I think the "plus one" is the whole substring, no?  Seems like there should be a better way to check this.

Reply:
This is just how this regex class works. The first match is the whole regex match and the other matches are the groups within the regex